src/prop.ts

```diff
@@ -7,6 +7,13 @@
  * @data_last
  * @category Object
  */
-export function prop<K extends string = string>(name: K) {
-  return <T extends Record<any, any>>(obj: T) => obj[name];
+
+interface PropInterface {
+  <T, K extends keyof T>(name: K): (obj: T) => T[K];
+  <K extends string = string>(name: K): <T extends Record<any, any>>(
```

Comment:
Can this just be ʼobjectʼ instead of record? Its a little more broad and correct (for example you can use prop("length“) on arrays.

If not i prefer the Record<PropertyKey, unknown> syntax as it's more eslint friendly